src/App.js

```diff
@@ -1,19 +1,295 @@
+    movieInfo: null
+  });
+  const [movieFormResult, setMovieFormResult] = useState({
+    msg: null,
+    isOpen: false,
+    isSuccess: true
+  })
+
+  const openDeleteMovieDialog = (movieInfo) => {
+    setDeleteMovieDialog({
+      isOpen: true,
+      movieInfo: movieInfo
+    });
+  };
+
+  const closeDeleteMovieDialog = () => {
+    setDeleteMovieDialog({
+      isOpen: false,
+      movieInfo: null
+    });
+  };
+
+  const openMovieFormResultDialog = (msg, isSuccess) => {
+    setMovieFormResult({
+      isSuccess: isSuccess,
+      msg: msg,
+      isOpen: true
+    })
+  };
+
+  const closeMovieFormResultDialog = () => {
+    setMovieFormResult({
+      isSuccess: true,
+      msg: null,
+      isOpen: false
+    })
+  };
+  
+
+  useEffect(() => {
+    console.log("useEffect Setting Genre Menu's", selectedGenre, fetchCurrentQueryParams(searchParams))
+    let tempList = [];
+    fetchGenreList()
+    .then((response) => {
+      response.data.data.map(i=>{
+        tempList = tempList.concat(i.genres)
+        return i.genres;
+      });
+      tempList = Array.from(new Set(tempList));
+      setGenreList(tempList.toSorted())
+      return tempList;
+    })
+    .catch((err) => {
+      console.log('error', err)
+     });
+  }, [])
+
+  
+  useEffect(() => {
+    updateCurrentURL()
+    console.log("useEffect called.>>")
+    callMoviewAPI(selectedGenre, sortBy, query);
+  }, [selectedGenre, sortBy, query, searchString])
+
+  const updateCurrentURL=()=>{
+    const queryParams ={activeGenre: selectedGenre, sortBy: sortBy};
+    if(searchString){
+      queryParams['query'] = searchString;
+    }
+    setSearchParams(queryParams);
+  }
+
+  const callMoviewAPI = (selectedGenre, sortBy,  searchString) =>{
+    let searchQuerParam = new URLSearchParams();
+    searchQuerParam.set("offset", 0);
+    searchQuerParam.set("limit", 20);
+    if(selectedGenre && !searchString){
+      searchQuerParam.set("filter", selectedGenre);
+    }
+    searchQuerParam.set("sortBy", sortBy);
+    searchQuerParam.set("sortOrder", "desc");
+    if (searchString) {
+      searchQuerParam.set("searchBy", "title");
+      searchQuerParam.set("search", searchString);
+    }
+    axios.get(HOST+"/movies?"+searchQuerParam.toString())
+    .then((response) => {
+      setMovies(response.data.data);
+    })
+    .catch((err) => {
+      console.log('error', err)
+     });
+  }
+
+  const createSearchParamsCustom = ()=>{
+    const queryParams ={activeGenre: selectedGenre, sortBy: sortBy};
+    if(searchString){
+      queryParams['query'] = searchString;
+    }
+    return createSearchParams(queryParams);
+  }
+  const onGenreSelected = (selectedGenre) => {
+    console.log("You selected: "+ selectedGenre)
+    setSearchString(null)
+    setSelectedGenre(selectedGenre);
+    navigate({
+      pathname: "/",
+      search: createSearchParamsCustom().toString()
+    });
+  }
+
+  const onTileSelected = (movieData) => {
+    navigate("/movie/"+movieData.id)
```

Comment:
```suggestion
    navigate(`/movie/${movieData.id}`)
```
Always use string interpolation to concatenate the strings